src/main.rs

```diff
@@ -282,8 +282,43 @@ struct CheckRelease {
     verbosity: clap_verbosity_flag::Verbosity<clap_verbosity_flag::InfoLevel>,
 }
 
-#[test]
-fn verify_cli() {
-    use clap::CommandFactory;
-    Cargo::command().debug_assert()
+#[cfg(test)]
+mod tests {
```

Comment:
> thread 'tests::verify_binary_contains_lints' panicked at 'called `Result::unwrap()` on an `Err` value: CargoError { cause: Some(NotFoundError { path: "/home/runner/work/cargo-semver-checks/cargo-semver-checks/target/debug/cargo-semver-checks" }) }', src/main.rs:298:69

I think the guarantee that the binary is built first is only available for "integration" / "black box" tests (in the `tests/` directory)